File: 1523-count-odd-numbers-in-an-interval-range/1523-count-odd-numbers-in-an-interval-range.py
class Solution:
    def countOdds(self, low: int, high: int) -> int:
        diff = high - low
        if not(diff):
            return 1 if (low % 2) == 1 else 0
        if diff % 2 == 0:
            return (diff // 2) + 1 if (low % 2) == 1 else diff // 2
        return (diff // 2) + 1

File: 1523-count-odd-numbers-in-an-interval-range/1523-count-odd-numbers-in-an-interval-range.rs
impl Solution {
    pub fn count_odds(low: i32, high: i32) -> i32 {
        let diff: i32 = high - low;
        if (diff == 0) {
            if ((low % 2) == 1) {
                return 1;
            }
            return 0;
        }
        if ((diff % 2) == 0) {
            if ((low % 2) == 1) {
                return (diff / 2) + 1;
            }
            return diff / 2;
        }
        return (diff / 2) + 1;
    }
}